Function/BookFunc/Services/BookingService.cs

```diff
@@ -0,0 +1,131 @@
+using System.Runtime.CompilerServices;
+using TeamsMeetingBookFunc.Models;
+using TeamsMeetingBookFunc.Helpers;
+using Microsoft.Graph.Extensions;
+
+namespace TeamsMeetingBookFunc.Services
+{
+    class BookingService
+    {
+        private readonly IConfigurationRoot configuration;
+        private readonly GraphServiceClient graphServiceClient;
+
+        public BookingService(IConfigurationRoot configuration)
+        {
+            this.configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+
+            var authProvider = GetAuthenticationProvider(configuration);
+
+            graphServiceClient = new GraphServiceClient(authProvider);
+        }
+
+        internal async Task<Event> CreateCalendarEventAsync(RequestModel requestModel, string bodyText)
+        {
+            var attendeeList = new List<Attendee>();
+
+            var newEvent = new Event
+            {
+                Subject = requestModel.Subject,
+                Body = new ItemBody
+                {
+                    ContentType = BodyType.Text,
+                    Content = bodyText
+                },
+                Attendees = attendeeList,
+                Start = requestModel.StartDateTime.Value.ToDateTimeTimeZone(),
+                End = requestModel.EndDateTime.Value.ToDateTimeTimeZone(),
+            };
+
+            if (!String.IsNullOrWhiteSpace(requestModel.DoctorEmailAddress))
+            {
+                attendeeList.Add(new Attendee
+                {
+                    EmailAddress = new EmailAddress
+                    {
+                        Address = requestModel.DoctorEmailAddress
+                    },
+                    Type = AttendeeType.Required
+                }
+                );
+            }
+
+            if (!String.IsNullOrWhiteSpace(requestModel.PatientEmailAddress))
+            {
+                attendeeList.Add(new Attendee
+                {
+                    EmailAddress = new EmailAddress
+                    {
+                        Address = requestModel.PatientEmailAddress
+                    },
+                    Type = AttendeeType.Required
+                }
+                );
+            }
+
+            foreach (var emailAddress in requestModel.OptionalAttendees)
+            {
+                attendeeList.Add(new Attendee
+                {
+                    EmailAddress = new EmailAddress
+                    {
+                        Address = emailAddress
+                    },
+                    Type = AttendeeType.Resource
+                }
+                );
+            }
+
+            return await graphServiceClient.Me.Events.Request()
+                .AddAuthenticationToRequest(configuration.GetConnectionStringOrSetting(ConfigConstants.UserEmailCfg), configuration.GetConnectionStringOrSetting(ConfigConstants.UserPasswordCfg))
+                .WithMaxRetry(5)
+                .AddAsync(newEvent).ConfigureAwait(false);
```

Comment:
This requires Calendars.ReadWrite permission on the Microsoft Graph. We must update the wiki before merging.
Wiki page to update: https://github.com/saguar/TeamsMeetingsBooking/wiki/Deploy-and-Configuration
Section to update: the one with the 'az ad app permission add' command

Reply:
I suggest to split these feature in a different commits.